### DISCUSSION TITLE
[QUESTION] When is SKGLView.CanvasSize set

### DISCUSSION THREAD

**Discussion by charlenni** [mono/SkiaSharp#1648] [Q&A]
I use the CanvasSize to calculate the scaling from Forms independent coordinates to device dependent coordinates. But if I try to get the CanvasSize in OnSizeChanged, it is always (0, 0), or better, is the last size of the view. It seems, that I have to check for a change in OnPaintSurface. But I don't want to check this each 16 ms, because it doesn't change without a OnSizeChanged event.

So the question is: Is there an event to get the correct CanvasSize?

**Comment by mattleibow** [comment]
Unfortunately this is a current limitation of the view because the canvas size is only calculated just before a frame is rendered.

However, this is also not a complex system and I probably will change this to be pre-calculated. You can get the next frame size by basically multiplying the forms view bounds by the screen density.

**Reply by charlenni** [reply]
Thank you for your time to looking into this issue.

I now use the Xamarin.Essential library and multiply `Width`/`Height` of the view with the `Xamarin.Essentials.DeviceDisplay.MainDisplayInfo.Density` to get the same result as using `CanvasSize.Width`/`CanvasSize.Height`.
